accepted/00000-hub-reporting.md

```diff
@@ -0,0 +1,177 @@
+- Feature Name: Hub Reporting
+- Start Date: 2021-11-23
+
+# Summary
+[summary]: #summary
+
+Design and implement the infrastructure for a central reporting on Uyuni Hub.
+
+# Motivation
+[motivation]: #motivation
+
+In a Uyuni Hub scenario we have the Hub to prepare and provide content for multiple other Uyuni Servers.
+The goal is now to get data from these Servers back and have combined reporting data available on the Hub.
+
+The data should be made available for external Reporting Tools.
+
+If possible, the data should also be usable on a single Uyuni Server with the same mechanism.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The data should be made available in a PostgreSQL database. We considered using other database types,
+but the increased maintenance effort for new packages and limited knowledge about NoSQL databases
+let us stay with the good well known PostgreSQL database.
+Also PostgreSQL seems to be accepted in most Reporting tools as data source, while NoSQL databases
+are not supported, or only via standard DBMS modules. Native support was rarely available.
+
+### The Database
+The main database is a PostgresDB in the Uyuni Hub system (but in the future it will be possible to 
+use also an external db): it stores all the information collected from all the server, and eventually
+aggregates them. Other databases with the same schema are also present in all the server, to collect 
+information for that system.
+All the databases (in the Uyuni Hub and in all Uyuni Server) needs to be made available on the network to either connect in a secure way (using the SSL
+certificates provided by Uyuni Server) with the reporting tool or (for Uyuni Server) with the Hub to gather the data. 
+If possible the connection to the main Uyuni DB from the outside should be forbidden.
+In summary:
+- DB in Uyuni Hub stores, collects and eventually aggregates data coming from all the DB in Uyuni Server,
+- DB in Uyuni Server stores it own data,
+- The reporting tool can be connected either to the Hub or to any Uyuni Server,
+- all connections are secure.
+
+### The Database Schema
+
+The schema should export the most important tables from the main Uyuni Database as a slightly de-normalized
+variant containing only data which are relevant for a report.
```

Comment:
I think it would be fair to say that main Uyuni DB schema is _very_ normalized - which is great from a DB design perspective, but generally difficult to write query against.

Thus, given the goals of this effort, I would have expected a statement like "heavily de-normalized" instead of "slightly" so - what trade-offs are you projecting to balance here?

What pros and cons do you see in denormalizing more or less?

Reply:
Example: de-normalizing "rhnServer". 

org_id => org name
creator_id => creator name
contact_method_id => method name

This is the kind of normalization we want to do. instead of using IDs to point to single values, we can write the values directly. We also want to remove unimportant values from the report.

But multi value fields like "network interfaces" or maybe virtualization functionality we want to provide via extra tables which still needs to be joined.

With this we let the customer the choise on "how" he want to make use of it in his report.

Not sure if we name it "slightly" or "heavy" de-normalized. Maybe we just say "de-normalized"

Reply:
OK, I think I agree in principle - the goal should be to try to reduce the table count significantly somehow. Let's iterate on the detailed schema when the time comes. Thanks!